src/app/portlets/content-types/fields/service/field-drag-drop.service.ts

```diff
@@ -214,21 +220,34 @@ export class FieldDragDropService {
     private shouldDrag(source: HTMLElement, isDragButton: boolean): boolean {
         return this.isDraggingFromSource(source) || isDragButton;
     }
-
+    // tslint:disable-next-line: cyclomatic-complexity
     private shouldAccepts(
-        _el: HTMLElement,
-        source: HTMLElement,
-        _handle: HTMLElement,
+        el: HTMLElement,
+        target: HTMLElement,
+        _source: HTMLElement,
         _sibling: HTMLElement
     ): boolean {
-        return source.dataset.dragType !== 'source';
+        console.log(el, target);
+        const columnsCount = target.parentElement.querySelectorAll('.row-columns__item').length;
+        const isColumnField = el.dataset.clazz === COLUMN_BREAK_FIELD.clazz;
+        const cantAddColumn = isColumnField && columnsCount >= 4;
+
+        if (cantAddColumn) {
+            this.clearCurrentFullRowEl();
+            this.currentFullRowEl = target.parentElement.parentElement;
+            this.currentFullRowEl.style.opacity = '0.4';
```

Comment:
use a class instead ?

Reply:
I think about it, but I want to unit tests that this happen, with a class I can assert the class show or not but not the styling.